openproblems/tasks/spatial_decomposition/datasets/utils.py

```diff
@@ -139,7 +139,7 @@ def generate_synthetic_dataset(
     adata_spatial.obsm["n_cells"] = sp_c
     adata_merged = merge_sc_and_sp(adata, adata_spatial, test=test)
     adata_merged.X[adata_merged.X == np.inf] = adata_merged.X.max()  # remove inf
-    adata_merged.layers["counts"] = adata_merged.X.copy()
+    adata_merged.layers["counts"] = adata_merged.X
```

Comment:
why not have a deepcopy of the .X? If .X gets normalized otherwise, so does the counts layer, no?

Reply:
ah, okay. So normalizers should copy. I missed that, thx.